src/helix-ui/elements/HXModalElement.js

```diff
@@ -0,0 +1,77 @@
+import { HXElement } from './HXElement';
+import { KEYS } from '../util';
+import shadowStyles from './_hx-modal.less';
+
+const tagName = 'hx-modal';
+const template = document.createElement('template');
+
+template.innerHTML = `
+  <style>${shadowStyles}</style>
+  <div id="container">
```

Comment:
Better, but I think we can simplify even more.

```html
<button id="close">
  <hx-icon type="times"></hx-icon>
</button>
<slot></slot>
```

Reply:
I've run into multiple issues trying to remove the container div here and apply all the modal styles to the light DOM. Namely, I haven't found a clean way to both position the modal in the center of the screen, while styling the backdrop within `:before`.

The best way I've found to position the modal is to use `translate`, since the modal can vary in size. This causes the children of the modal, like the `:before` backdrop element to also be translated, so it's tricky to get that element to cleanly cover the whole screen. I've found unintuitive ways but nothing I really like.

Reply:
Tinkering around with it myself, I've come to the same conclusion. This is acceptable for now. We'll revisit it again later.